### ISSUE TITLE
Salsa does not warn or help when passing in wrong argument type

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/vscode#2442]
Testing #2218 

```javascript
/**
 * @param {string} name
 */
function howdy(name) {
    
}

howdy(3);
```

![image](https://cloud.githubusercontent.com/assets/5047891/12614145/559c1f9e-c4fe-11e5-9591-3f2273015a46.png)

even if it knows a string is expected:

![image](https://cloud.githubusercontent.com/assets/5047891/12614165/7babf66e-c4fe-11e5-9d3f-97ca98bd11bf.png)

**Comment by billti** [comment]
Correct. We explicitly decided to give no warnings/errors other than syntax errors for JavaScript code. If warnings on incorrect type usage are required, we'd need to spec how and when these would occur in JavaScript, else being based on TypeScript, code such as the below would also be a type error:

```
var x = 10;
var x = "ten";

var y = {a:true};
y.b = false;
```

**Comment by egamma** [comment]
In the existing Code JS language service we gave warnings for the first example above and we didn't receive complaints.

![errors](https://cloud.githubusercontent.com/assets/172399/12623429/6f551a9a-c52a-11e5-87bb-98d3e0830fa8.png)

We have also warned about undefined variables, which had issues when users where also using a linter in their project and they defined a global as a global in their linter configuration. The confusion was that they define the global in linter configuration, but our JS service still warned about the global.
